traitsui/tests/_tools.py

```diff
@@ -67,33 +68,102 @@ def handle_exception(object, trait_name, old, new):
             raise exceptions[0]
 
 
+# Toolkit constants
+
+class ToolkitName(enum.Enum):
+    wx = "wx"
+    qt = "qt"
+    null = "null"
+
+
+def is_wx():
+    """ Return true if the toolkit backend is wx. """
+    return ETSConfig.toolkit == ToolkitName.wx.name
+
+
+def is_qt():
+    """ Return true if the toolkit backend is Qt
+    (that includes Qt4 or Qt5, etc.)
+    """
+    return ETSConfig.toolkit.startswith(ToolkitName.qt.name)
+
+
+def is_null():
+    """ Return true if the toolkit backend is null.
+    """
+    return ETSConfig.toolkit == ToolkitName.null.name
+
+
+def requires_toolkit(toolkits):
+    """ Decorator factory for skipping tests if the current toolkit is not
+    one of the given values.
+
+    Parameters
+    ----------
+    toolkits : iterable of members of ToolkitName
+        e.g. ``list(ToolkitName)`` to include all toolkits.
+    """
+    mapping = {
+        ToolkitName.null: is_null,
+        ToolkitName.qt: is_qt,
+        ToolkitName.wx: is_wx,
+    }
+    return skipIf(
+        not any(mapping[toolkit]() for toolkit in toolkits),
+        "Test requires one of these toolkits: {}".format(toolkits)
+    )
+
+
+def _deprecated(func):
+    # Emit warning to ease tension across pending pull requests.
+    # This function should be removed when no more active PRs are relying
+    # on this.
+    def wrapped(*args, **kwargs):
+        warnings.warn(
+            "{!r} will be removed. "
+            "Use is_wx, is_qt, is_null or requires_toolkit instead.".format(
+                func
+            ),
+            DeprecationWarning,
+            stacklevel=3,
+        )
+        return func(*args, **kwargs)
+    return wrapped
```

Comment:
Could you explain how this function works after the standup today?

Reply:
(Discussed offline)